apps/shell/tash_main.c

```diff
@@ -156,9 +167,18 @@ char *tash_read_input_line(int fd)
 				shdbg("TASH: can not read uart\n");
 				return buffer;
 			}
-
+#ifdef CONFIG_PM
+			pmdrv_fd = open(PM_DRIVER, O_WRONLY);
+			if (pmdrv_fd < 0) {
+				shdbg("open %s failed(%d), \n", PM_DRIVER, get_errno());
+			}
+#endif
 			for (char_idx = 0; char_idx < nbytes; char_idx++) {
-
+#ifdef CONFIG_PM
+				if ((pmdrv_fd >= 0) && (ioctl(pmdrv_fd, PMIOC_TIMEDSUSPEND, &pm_suspend_arg) != OK)) {
```

Comment:
At this point, we have multiple characters in our buffer and we are processing one character at a time. I dont think we need to suspend the pm here. The pm suspend must be called before read/poll call, it at the first line of the do-while loop.

Reply:
Done